service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/PlacementSubmittedControllerTest.java

```diff
@@ -0,0 +1,146 @@
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(PlacementController.class)
+@OverrideAutoConfiguration(enabled = true)
+class PlacementSubmittedControllerTest extends AbstractControllerTest {
+    @MockBean
+    private NotificationClient notificationClient;
+    @MockBean
+    private CoreCaseDataService coreCaseDataService;
+
+    private static final Long CASE_ID = 12345L;
+    private static final String CASE_REFERENCE = "12345";
+    private static final String RESPONDENT_SURNAME = "Watson";
+    private static final String LOCAL_AUTHORITY_CODE = "example";
+    private static final String SEND_DOCUMENT_EVENT = "internal-change:SEND_DOCUMENT";
+
+    private final DocumentReference documentReference = DocumentReference.builder().build();
+
+    PlacementSubmittedControllerTest() {
+        super("placement");
+    }
+
+    @Test
+    void shouldSendNotificationWhenAddingNewChildPlacement() throws Exception {
+        Element<Child> child1 = testChild();
+        Element<Child> child2 = testChild();
+
+        DocumentReference child1Application = testDocument();
+        DocumentReference child2Application = testDocument();
+
+        Element<Placement> child1Placement = element(testPlacement(child1, child1Application));
+        Element<Placement> child2Placement = element(testPlacement(child2, child2Application));
+
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(CASE_ID)
+                .data(ImmutableMap.<String, Object>builder()
+                    .putAll(buildNotificationData())
+                    .putAll(buildPlacementData(List.of(child1, child2), List.of(child2Placement, child1Placement),
+                        child2.getId()))
+                    .build())
+                .build())
+            .caseDetailsBefore(CaseDetails.builder().data(new HashMap<>()).build())
+            .build();
+
+        postSubmittedEvent(callbackRequest);
+
+        verify(notificationClient).sendEmail(
+            eq(NEW_PLACEMENT_APPLICATION_NOTIFICATION_TEMPLATE), eq("admin@family-court.com"),
+            eq(expectedTemplateParameters()), eq(CASE_REFERENCE));
+        verify(coreCaseDataService).triggerEvent(null, null, CASE_ID, SEND_DOCUMENT_EVENT, Map.of(
+            "documentToBeSent", documentReference
+        ));
+    }
+
+    @Test
+    void shouldNotSendANotificationWhenUploadingPreviousChildPlacement() throws Exception {
+        Element<Child> child1 = testChild();
+        DocumentReference child1Application = testDocument();
+        Element<Placement> child1Placement = element(testPlacement(child1, child1Application));
+
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(CASE_ID)
+                .data(buildPlacementData(List.of(child1), List.of(child1Placement), child1.getId()))
+                .build())
```

Comment:
can case details be build once and passed to both places, it would be obvious that they are same. But tbh, test should verify that nothing is send when document stay same

Reply:
Good point, extracted common `CaseDetails` to variable.